compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/Utils.java

```diff
@@ -438,4 +450,30 @@ public static Optional<CodeActionInfo> getCodeActionInfo(CodeActionContext codeA
                 diagnostic.location().lineRange());
         return Optional.of(CodeActionInfo.from(codeAction, List.of(locationArg)));
     }
+
+    private static boolean isCloseFrameRecordType(TypeSymbol typeSymbol) {
+        if (typeSymbol instanceof TypeReferenceTypeSymbol) {
+            if (typeSymbol.nameEquals(CLOSE_FRAME) &&
+                    typeSymbol.getModule().flatMap(Symbol::getName).orElse("").equals(PACKAGE_WEBSOCKET)) {
+                return true;
+            }
+            typeSymbol = ((TypeReferenceTypeSymbol) typeSymbol).typeDescriptor();
+            return isCloseFrameRecordType(typeSymbol);
+        } else if (typeSymbol instanceof RecordTypeSymbol bRecordTypeSymbol) {
+            if (bRecordTypeSymbol.fieldDescriptors().containsKey(CLOSE_FRAME_TYPE)) {
+                TypeSymbol objectType = bRecordTypeSymbol.fieldDescriptors().get(CLOSE_FRAME_TYPE).typeDescriptor();
+                String moduleName = objectType.getModule().flatMap(Symbol::getName).orElse("");
+                return moduleName.equals(PACKAGE_WEBSOCKET) &&
+                        (objectType.nameEquals(PREDEFINED_CLOSE_FRAME_TYPE) ||
+                                objectType.nameEquals(CUSTOM_CLOSE_FRAME_TYPE));
+            }
+        } else if (typeSymbol instanceof IntersectionTypeSymbol intersectionTypeSymbol) {
+            for (TypeSymbol memberType : intersectionTypeSymbol.memberTypeDescriptors()) {
```

Comment:
I feel like, this implementation is wrong.

Lets check the readonly thing separately.
Then lets get the `effectiveType` and check this property.